src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -108,53 +279,62 @@ private void launchNode(SchedulerDriver driver, Offer offer,
-                String.format("http://%s:%d/%s", conf.getFrameworkHostAddress(), confServerPort,
-                    HDFSConstants.HDFS_BINARY_FILE_NAME))
-                .build(),
-            CommandInfo.URI.newBuilder().setValue(
-                String.format("http://%s:%d/%s", conf.getFrameworkHostAddress(), confServerPort,
-                    HDFSConstants.HDFS_CONFIG_FILE_NAME))
-                .build()))
-            .setEnvironment(Environment.newBuilder()
-                .addAllVariables(Arrays.asList(
-                Environment.Variable.newBuilder()
-                    .setName("HADOOP_OPTS")
-                    .setValue(conf.getJvmOpts()).build(),
-                Environment.Variable.newBuilder()
-                    .setName("HADOOP_HEAPSIZE")
-                    .setValue(String.format("%d", conf.getHadoopHeapSize())).build(),
-                Environment.Variable.newBuilder()
-                    .setName("HADOOP_NAMENODE_OPTS")
-                    .setValue("-Xmx" + conf.getNameNodeHeapSize() + "m").build(),
-                Environment.Variable.newBuilder()
-                    .setName("HADOOP_DATANODE_OPTS")
-                    .setValue("-Xmx" + conf.getDataNodeHeapSize() + "m").build(),
-                Environment.Variable.newBuilder()
-                    .setName("EXECUTOR_OPTS")
-                    .setValue("-Xmx" + conf.getExecutorHeap() + "m").build())))
-                    .setValue(
-                        "env ; cd hdfs-mesos-* && exec java $HADOOP_OPTS $EXECUTOR_OPTS " +
+        .setCommand(
+            CommandInfo
+                .newBuilder()
+                .addAllUris(
+                    Arrays.asList(
+                        CommandInfo.URI
+                            .newBuilder()
+                            .setValue(
+                                String.format("http://%s:%d/%s", conf.getFrameworkHostAddress(),
+                                    confServerPort,
+                                    HDFSConstants.HDFS_BINARY_FILE_NAME))
+                            .build(),
+                        CommandInfo.URI
+                            .newBuilder()
+                            .setValue(
+                                String.format("http://%s:%d/%s", conf.getFrameworkHostAddress(),
+                                    confServerPort,
+                                    HDFSConstants.HDFS_CONFIG_FILE_NAME))
+                            .build()))
+                .setEnvironment(Environment.newBuilder()
+                    .addAllVariables(Arrays.asList(
+                        Environment.Variable.newBuilder()
+                            .setName("HADOOP_OPTS")
+                            .setValue(conf.getJvmOpts()).build(),
+                        Environment.Variable.newBuilder()
+                            .setName("HADOOP_HEAPSIZE")
+                            .setValue(String.format("%d", conf.getHadoopHeapSize())).build(),
```

Comment:
What about getJournalNodeHeapSize and getZkfcHeapSize? Looks like they're never used anywhere for actually defining task/jvm sizes.